### DISCUSSION TITLE
Missing solc installation

### DISCUSSION THREAD

**Discussion by AlbertoSinigaglia** [smartcontractkit/full-blockchain-solidity-course-py#5] [Q&A] [answered]
During the first compilation (around 3:45:00), you are not showing that is required to install the `solc` package to effectively compile the Solidity contract.

I've solved with:
```
import solcx
solcx.install_solc('0.8.7') # or whatever version we want
```

**Comment by oceanKingEth** [comment]
thank you Alberto! I got stuck here as well. this worked perfectly. I added in version ('0.6.0') to match the tutorial and it worked as well.

**Reply by RobustCelery** [reply]
Thanks for the help, this worked for me also

**Reply by alexpost81** [reply]
that doesn't work

**Reply by HalaAura** [reply]
That was really helpful!! thank you so much I got stuck there for two days but I tried 
import solcx
solcx.install_solc('0.8.7')  
and it worked. Thanks a million!

**Reply by AussieCryptoBen** [reply]
still not working tried for days. like solcx no longer supported. I am also think that the program is not linking the install to the command.

**Reply by cromewar** [reply]
Hello @Aussiecpto2 could you please provide more info about what you tried and may be screenshots about the errors you have?

**Reply by jt-berg-whitt** [reply]
not a single answer in the 5 different threads opened about this issue are working answers

**Reply by mr-ziash** [reply]
i found this solution the pip version (you type pip --version) in the terminal should be the same as the python version (python --version) and the python version in the bottom left corner, but I'm still getting that INFO: could not find files for the gives pattern(s), is that okay? also I don't see that bunch of code that was shown in the video I just get that info thingy

**Reply by cromewar** [reply]
@mr-ziash, this INFO: could not find files for the gives pattern(s) is not an error, so you have nothing to worry about ;)

**Reply by Rohann608** [reply]
I've tried the command that you suggested and it still throws an error. I've already installed the solcx using the command `pip install py-solc-x` and when I'm running the code it seems like vscode is not recognizing solcx import.

it gives out the error:
NameError: name 'compile_standard' is not defined
PS C:\Users\hp\web3_py_simple_storage> python -u "c:\Users\hp\web3_py_simple_storage\deploy.py"
INFO: Could not find files for the given pattern(s).
Traceback (most recent call last):
  File "c:\Users\hp\web3_py_simple_storage\deploy.py", line 8, in <module>
    compiled_sol = compile_standard(
NameError: name 'compile_standard' is not defined
PS C:\Users\hp\web3_py_simple_storage> 

the code that I'm running:
`import solcx

solcx.install_solc("0.6.0")

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {
                    "*": ["abi", "metadata", "evm.bytecode", "evm.bytecode.sourceMap"]
                }
            }
        },
    },
)
print(compiled_sol)
`


![image](https://user-images.githubusercontent.com/126192345/224470322-cc81280a-c26e-44ee-a473-48513b6041ee.png)

**Comment by PatrickAlphaC** [comment, marked as answer]
Thanks so much!!

I've added this to our issue tracker, and updated our code examples: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md#lesson-4

Thank you so much!

**Reply by iamtobaey** [reply]
@PatrickAlphaC I did exactly this but still get the same error and I'm stuck here.
This is the code 
```from solcx import compile_standard, install_solc
import json

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()
    print(simple_storage_file)

print("Installing...")
install_solc("0.6.0")
# compile our solidity

compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {
                    "*": ["abi", "metadata", "evm.bytecode", "evm.bytecode.sourceMap"]
                }
            }
        },
    },
    solc_version="0.6.0",
)
# print(compiled_sol)

# with open("compiled_code.json", "w") as file:
    # json.dump(compiled_sol, file)




### The Error
C:\Users\Ademola Tobaey\demos\web3_py_simple_storage>python deploy.py
Traceback (most recent call last):
  File "C:\Users\Ademola Tobaey\demos\web3_py_simple_storage\deploy.py", line 1, in <module>
    from solcx import compile_standard, install_solc
  File "C:\Users\Ademola Tobaey\AppData\Local\Programs\Python\Python39\lib\site-packages\solcx\__init__.py", line 27, in <module>
    set_solc_version(get_installed_solc_versions()[-1], silent=True)
  File "C:\Users\Ademola Tobaey\AppData\Local\Programs\Python\Python39\lib\site-packages\solcx\install.py", line 126, in set_solc_version
    get_executable(version, solcx_binary_path)
  File "C:\Users\Ademola Tobaey\AppData\Local\Programs\Python\Python39\lib\site-packages\solcx\install.py", line 117, in get_executable
    raise SolcNotInstalled(
solcx.exceptions.SolcNotInstalled: solc v0.6.0 has not been installed. Use solcx.install_solc('v0.6.0') to install.

**Reply by PatrickAlphaC** [reply]
`"language": "Solidity",`  -> `"language": "solidity",`

**Reply by JMiddey** [reply]
Use solcx.install_solc('v0.6.0') to install.
edit:
and:
import solcx

**Reply by jt-berg-whitt** [reply]
Still doesnt work:
`from solcx import compile_standard, install_solc

install_solc("0.6.0")


with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()
    print(simple_storage_file)

# Compile our Solidity

compiled_sol = compile_standard(
    {
        "language": "solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)

print(compiled_sol)
`

**Reply by jt-berg-whitt** [reply]
I've also highlighted all the code and `shift+enter` seems to make it run the code? Here's the printout:

`>>> from solcx import compile_standard, install_solc
>>> install_solc("0.6.0")
Version('0.6.0')
>>> print("Hopfully installing, but probably not...")
Hopfully installing, but probably not...
>>> with open("./SimpleStorage.sol", "r") as file:
...     simple_storage_file = file.read()
...     print(simple_storage_file)

... 
// SPDX-License-Identifier: MIT
pragma solidity 0.8.0;

contract SimpleStorage {
    // this initializes to 0 and internal
    uint256 favoriteNumber;

    struct People {
        uint256 favoriteNumber;
        string name;
    }

    People[] public people;
    mapping(string => uint256) public nameToFavoriteNumber;

    function store(uint256 _favoriteNumber) public {
        favoriteNumber = _favoriteNumber;
    }

    function retrieve() public view returns (uint256) {
        return favoriteNumber;
    }

    function addPerson(string memory _name, uint256 _favoriteNumber) public {
        people.push(People(_favoriteNumber, _name));
        nameToFavoriteNumber[_name] = _favoriteNumber;
    }
}

>>> compiled_sol = compile_standard(
...     {
...         "language": "solidity",
...         "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
...         "settings": {
...             "outputSelection": {
...                 "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
...             }
...         },
...     },
...     solc_version="0.6.0",
... )
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/solcx/main.py", line 394, in compile_standard
    raise SolcError(
solcx.exceptions.SolcError: Only "Solidity" or "Yul" is supported as a language.
> command: `/Users/jacobwhitt/.solcx/solc-v0.6.0 --standard-json`
> return code: `0`
> stdout:
{"errors":[{"component":"general","formattedMessage":"Only \"Solidity\" or \"Yul\" is supported as a language.","message":"Only \"Solidity\" or \"Yul\" is supported as a language.","severity":"error","type":"JSONError"}]}

> stderr:
>>> 
>>> print(compiled_sol)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'compiled_sol' is not defined
>>> `

**Reply by jt-berg-whitt** [reply]
Still it seems that I am getting the same error about language supported, but i wrote solidity in lower case as I saw other people answering to do.
I also have this error "compiled_sol isn't defined" even though it def is...
all this on top of the console showing me this:
python --version // Python 2.7.16 < this seems to be the base version in my Mac OS, even though I chose 3.10 which I just installed...I even chose multiple versions of Python in the bottom left corner of VSCode to no avail

**Reply by ElyWacime** [reply]
i keep getting this error :
raise ValueError('Invalid version string: %r' % version_string)
ValueError: Invalid version string: "solcx.install_solc('0.6.0')"

any idea how to solve this problem?
i asked chatGPT, and it says that i should run this commands:

sudo apt-get install software-properties-common
sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 9BA28146F7AD7FC9
sudo add-apt-repository "deb http://ppa.launchpad.net/ethereum/ethereum/ubuntu xenial main"

but after doing that i get this error:
 File "/usr/lib/python3/dist-packages/softwareproperties/ppa.py", line 126, in lpppa
    self._lpppa = self.lpteam.getPPAByName(name=self.ppaname)
                  ^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/softwareproperties/ppa.py", line 113, in lpteam
    self._lpteam = self.lp.people(self.teamname)
                   ^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'people'

**Reply by cromewar** [reply]
@ElyWacime I will recommend you to use an isolated python environment, and also check out your code as it seems you are referring to people when it's not able to find it.

**Comment by pinkCosmos** [comment]
Thank you for bringing up the problem, but I still get 

> bash: syntax error near unexpected token `'0.6.0''

 when i try to install the package.. 
With `npm install -g solc` and a rooted shell I was then able to successfully install it. But I am still not able to run the code, because i get the following error message every time I try: 

>   File "deploy.py", line 1, in <module>
>     from solcx import compile_standard
> ImportError: No module named solcx

Any Ideas what the Problem could be?

**Reply by PatrickAlphaC** [reply]
Can you place all the code of your script so far here?

**Reply by Evancitoo** [reply]
me to
![Screenshot_1](https://user-images.githubusercontent.com/67725330/137562627-011a80ad-1825-418e-8adb-768426a5c31c.jpg)
o

**Reply by Evancitoo** [reply]
![Screenshot_2](https://user-images.githubusercontent.com/67725330/137562701-e9277919-3185-483a-9e9e-8d545564435e.jpg)
3 things missing import. web3, solcx, dotevn. can not deploy the example =(.:!!

**Reply by PatrickAlphaC** [reply]
Did you do `pip install web3` `pip install py-solcx` `pip install python-dotenv`?

**Reply by AlfonsoJohn** [reply]
I have  ```pip install web3``` ```pip install py-solc-x``` ```pip install python-dotenv``` all installed and I copied the code exactly from Github, I'm still getting those import errors.

**Reply by cromewar** [reply]
hello @AlfonsoJohn please refer to this thread: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/209#discussioncomment-1499503

**Reply by AlfonsoJohn** [reply]
Hello @cromewar, hmmm I don't think I'm following, I'm not even using brownie nor ```pipx```. I did try using brownie and I got the same import not resolved error.

**Reply by cromewar** [reply]
@AlfonsoJohn are you selecting the proper python interpreter on vscode, on the bottom left corner you can select it.

**Reply by AlfonsoJohn** [reply]
@cromewar Not sure, which one do you recommend? These are the ones that show up for me.

<img width="618" alt="Screen Shot 2021-11-10 at 10 55 33 AM" src="https://user-images.githubusercontent.com/67883631/141147019-7b0988da-12d0-4067-8e01-bca84d16dd6d.png">

**Reply by cromewar** [reply]
You must select the one you have installed web3, solc and brownie if I have to guess it would be the the 3.10 version, but I don't know give it a try.

**Reply by SubratSahai** [reply]
its still not working after changing the interepreter too

**Reply by SubratSahai** [reply]
its not showing error but it still says solcx is not recogonised

**Comment by Bawera17** [comment]
Hello I have this error.    
 solc_version="0.6.0",  
    ^
SyntaxError: invalid syntax

My code is this:
```python
from solcx import compile_standard, install_solc

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()
    print(simple_storage_file)

install_solc("0.6.0")
compiled_sol = compile_standard(
    {
    "language": "Solidity", 
    "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
    "settings": {
        "outputSelection": {
            "*" : {"*": ["abi", "metadata", "evm.bitecode", "evm.sourceMap"]}
            }
        }
    }
    solc_version="0.6.0",
)

print(compiled_sol)
```
Thanks in advance.

**Reply by 8alpreet** [reply]
You're missing a comma after the closing curly brace of the first argument for the compile_standard func. 

Your code should look like:

```python
from solcx import compile_standard, install_solc

install_solc("0.6.0")

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)

print(compiled_sol)
``` 

ps: you can use three backticks to create code blocks so your code is easier to read. ([Link](https://www.freecodecamp.org/news/how-to-format-code-in-markdown/))

pps: I had to look up the backtick thing as well lol

**Reply by PatrickAlphaC** [reply]
Nice!!!!

**Reply by PatrickAlphaC** [reply]
Another tip, if you write `python` after the 3 tips, it might highlight it too!

**Reply by 8alpreet** [reply]
that's amazing!

**Reply by eitse** [reply]
raise SolcError("Could not determine the solc binary version")
solcx.exceptions.SolcError: Could not determine the solc binary version
> command: ``
> return code: `None`
> stdout:
None
> stderr:
None

Pls Senior Developers, i keep trying the exact updated code by Sir PatrickAlphaC and later that of Sir bhehar but still get same error as above. Your help could save my frustration :). Thanks

**Reply by eitse** [reply]
I Changed Solidity Version For my SimpleStorage.sol and deploy.py from 0.6.0, to 0.8.1, also tested with 0.8.1. then Created the compile_code.json. I think error was as a result of reiterating print(compiled_sol) by running python deploy.py multiple times after main.py is already created in solcx folder.

**Reply by 8alpreet** [reply]
Glad you got it sorted out!

**Reply by Anu-srinivasan** [reply]
Hi all,
my code is
from solcx import compile_standard, install_solc # <- import the install_solc method!



print ("installing...")

install_solc("0.6.0") # <- Add this line and run it at least once!
with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()
compiled_sol = compile_standard(
    {
        "language": "solidity",
        "sources": {"SimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)


Error is 
C:\Users\RBI\demos\simple_storage>python deploy.py
INFO: Could not find files for the given pattern(s).
installing...
Traceback (most recent call last):
  File "C:\Users\demos\simple_storage\deploy.py", line 10, in <module>
    compiled_sol = compile_standard(
  File "C:\Python39\lib\site-packages\solcx\main.py", line 394, in compile_standard   
    raise SolcError(
solcx.exceptions.SolcError: Only "Solidity" or "Yul" is supported as a language.      
> command: `C:\Users\.solcx\solc-v0.6.0\solc.exe --standard-json`
> return code: `0`
> stdout:
{"errors":[{"component":"general","formattedMessage":"Only \"Solidity\" or \"Yul\" is 
supported as a language.","message":"Only \"Solidity\" or \"Yul\" is supported as a language.","severity":"error","type":"JSONError"}]}

> stderr:

Kindly let me where I have gone wrong and what needs to be done by  me please

**Reply by AlbertoSinigaglia** [reply]
Most pf the times the error tells you everything, in this case _ Only "Solidity" or "Yul" is supported as a language_.  
Infact as you can see from here https://github.com/PatrickAlphaC/web3_py_simple_storage/blob/main/deploy.py you have to use `"language": "Solidity",` instead of `"language": "solidity",` @Anu-srinivasan

**Reply by jt-berg-whitt** [reply]
changed Solidity and still have issues. but if hitting "SHIFT + ENTER" - the code runs as in the tutorial video, but if i leave that terminal shell back to my main zsh shell, and type "python deploy.py" i get the "solcx module not found/installed" errors still... what gives?

**Reply by cromewar** [reply]
@jwhitty1  don't forget to select the correct python interpreter on vscode (bottom left corner) you should choose the one you have installed solc in, and of course open a new terminal.

**Comment by claxtonh** [comment]
I put in the fixes as suggested (from solcx import compile_standard, install_solc) and (install_solc("0.6.0").
The code appears to run correctly, EXCEPT in my terminal window I have 1 Problems.  When I click to view the problem it says,
"Unable to import 'solcx' pylint(import-error) [1,1]"

It still seems to run correctly, and I'm willing to just ignore it and continue on, though I'm a little worried it will cause issues down the road.  My code is below, in case you see something of concern.
```Solidity
`from solcx import compile_standard, install_solc


with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()
    print(simple_storage_file)

#compile Solidity

install_solc("0.6.0")
compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {
            "SimpleStorage.sol": {
                "content": simple_storage_file
            }
        },
        "settings": {
            "outputSelection": 
            {
                "*" : {
                    "*": [
                        "abi", "metadata", "evm.bytecode", "evm.sourceMap"
                    ]
                }
            }
        }
    },
    solc_version = "0.6.0"
)

print(compiled_sol)`
```

**Reply by cromewar** [reply]
hi @claxtonh yo can refer to [this](https://www.youtube.com/watch?v=BhNDn-FkKiU) video.

**Reply by claxtonh** [reply]
Awesome, that worked!  Thank you

**Reply by cromewar** [reply]
I'm glad is solved now.

**Reply by ghost** [reply]
First of all thank you cromewar,  It solved my importing problem thanks. 
I changed the interpreter to (windows store python) & it fixed it - I just picked the python with the windows store local file. hope it won't be an issue to what is coming nect in the course :) 
![Screenshot (9)](https://user-images.githubusercontent.com/96797534/147947298-91ab9232-e89a-4813-89a4-43077b8628d7.png)

**Comment by kingsingh21** [comment]
```Python
from solcx import compile_standard, install_solc

with open("./SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()
    print(simple_storage_file)

#compile Solidity

install_solc("0.6.0")
compiled_sol = compile_standard(
    {
        "language": "solidity",
        "sources": {
            "SimpleStorage.sol": {
                "content": simple_storage_file
            }
        },
        "settings": {
            "outputSelection": 
            {
                "*" : {
                    "*": [
                        "abi", "metadata", "evm.bytecode", "evm.sourceMap"
                    ]
                }
            }
        }
    },
    solc_version = "0.6.0"
)

print(compiled_sol)
```
========
Getting below error.

Traceback (most recent call last):
  File "C:\Users\User\demos\web3_py_simple_storage\deploy.py", line 13, in <module>
    compiled_sol = compile_standard(
  File "C:\Users\User\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\solcx\main.py", line 373, in compile_standard
    stdoutdata, stderrdata, command, proc = wrapper.solc_wrapper(
  File "C:\Users\User\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\solcx\wrapper.py", line 98, in solc_wrapper
    solc_version = _get_solc_version(solc_binary)
  File "C:\Users\User\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\solcx\wrapper.py", line 17, in _get_solc_version
    stdout_data = subprocess.check_output([str(solc_binary), "--version"], encoding="utf8")
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2544.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 424, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2544.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 528, in run       
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['C:\\Users\\User\\.solcx\\solc-v0.6.0\\solc.exe', '--version']' returned non-zero exit status 3221225595

please help.

**Reply by cromewar** [reply]
hello @kingsingh21 Solidity must be upper case.

**Reply by kingsingh21** [reply]
I am getting the same error even after changing "solidity" to "Solidity".

I just ran below code to see any issue

from solcx import compile_standard, install_solc

Getting below error
INFO: Could not find files for the given pattern(s).

**Reply by cromewar** [reply]
About `INFO: Could not find files for the given pattern(s).`is not an error so you don't have to worry about.

regarding to the problem above I recommend one of this options.

1. install a different solcx version.
2. use WSL.

**Comment by widmung89** [comment]


**Reply by cromewar** [reply]
Hello @widmung89 you have `setting` and the correct keyword is `settings`, that's why it isn't working. 
Give it a try ;)

**Reply by widmung89** [reply]
oh yes I saw that, that's why I remove my post! Thanks @cromewar !

**Comment by ashish060316** [comment]
> During the first compilation (around 3:45:00), you are not showing that is required to install the `solc` package to effectively compile the Solidity contract.
> 
> I've solved with:
> 
> ```
> import solcx
> solcx.install_solc('0.8.7') # or whatever version we want
> ```

Resolved the issue thanks

**Reply by selinimsibiri** [reply]
when i try to do that is says :  bash: syntax error near unexpected token `'0.8.7''
PLEASE HELP ME I HAVE BEEN TRYING TO FIX JUST THİS ERROR FOR 2 FRIKKING HOURS IM ABOUT TO CRY PLEASE HELP ME!!

**Reply by cromewar** [reply]
@Rinyata could you please share more info about the error, the code you used and also info about your system (windows, mac o linux)

**Reply by selinimsibiri** [reply]
> @Rinyata could you please share more info about the error, the code you used and also info about your system (windows, mac o linux)


Of cource:
 

i tried to change my python select interpere i also uninstalled solc and installed again. Also tried installing 1.1.1 and 1.1.0 versions but still im taking this error:

```
Traceback (most recent call last):
File "deploy.py", line 1, in
from solcx import compile_standard
ImportError: No module named solcx
```


can you please help me??
and here is my code:


```Python
from solcx import compile_standard
from solcx.install import install_solc

with open("./SimpleStorage.sol", "r") as file:
simple_storage_file = file.read()
print(simple_storage_file)

# Compile Our Solidity install_solc("0.6.0") compiled_sol = compile_standard( { "language": "Solidty", "sources": {"SimpleStorage.sol": {"content": simple_storage_file}}, "settings": { "outputSelection": { "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]} } }, }, solc_version="0.6.0", ) import solcx solcx.install_solc("0.8.7") print(compiled_sol)


```

**Reply by cromewar** [reply]
Well it seems you installation is unable to locate solcx, there are some things you could try.

1. use a python virtualenv, this allows you to manage all the libraries separately.
2. Do not forget to install `py-solc-x`.
3. if you are on windows try using WSL.

**Reply by selinimsibiri** [reply]
thank you so much ill try them now

**Reply by ashish060316** [reply]
import solcx
solcx.install_solc("0.8.7")
from solcx import compile_standard

try this. I am using mac and it worked.

**Reply by selinimsibiri** [reply]
i pasted this to my terminal :

python3 -m venv env 
source env/bin/activate
pip3 install py-solc-x 
pip3 install py-solc

it created a virtual enviroment and it worked . I hope it works for you too.

**Reply by AlrohfAZ** [reply]
THANK YOU!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by oliveruf42** [comment]
```Solidity
from solcx import compile_standard, install_solc  # <- Import the install_solc method!
install_solc("0.6.0")

with open("SimpleStorage.sol", "r") as file:
    simple_storage_file = file.read()

compiled_sol = compile_standard(
    {
        "language": "Solidity",
        "sources": {"ScleaimpleStorage.sol": {"content": simple_storage_file}},
        "settings": {
            "outputSelection": {
                "*": {"*": ["abi", "metadata", "evm.bytecode", "evm.sourceMap"]}
            }
        },
    },
    solc_version="0.6.0",
)
print("compiled_sol")

```

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.
Eddies-iMac:web3_py_simple_storage eddiebraddock$ python deploy.py
Traceback (most recent call last):
  File "deploy.py", line 1, in <module>
    from solcx import compile_standard, install_solc  # <- Import the install_solc method!
ImportError: No module named solcx
Eddies-iMac:web3_py_simple_storage eddiebraddock$ python3 deploy.py
compiled_sol
Eddies-iMac:web3_py_simple_storage eddiebraddock$

**Reply by oliveruf42** [reply]
I'm having an issue also. I've been stuck here for a couple of days. I've tried every suggestion on Stack Overflow. Is it something in my code or did i set up my env the wrong way ? I either get ImportError: No module named solcx 
or the terminal in VS Code seems to ignore all the other code and simply prints 
compiled_sol

**Reply by cromewar** [reply]
are you selecting the correct python interpreter on vscode (bottom left corner)?

**Reply by oliveruf42** [reply]
When i pull up the Python Interpreter in VS Code it gives me 3 options
# Python 3.10.1 64 bit /usr/local/bin/python3
# Python 3.8.9 64 bit /usr/bin/python3
# Python 3.10.1 64 bit ('env': venv) ./env/bin/python
The last one is from the virtual enviroment i setup as suggested on Stack Overflow. I've tried all three with no luck. Is there one i should try to add? (Thank you for helping me)

**Reply by oliveruf42** [reply]
It's not throwing Errors anymore, but now it's like the interpreter is ignoring all the code above print("compiled_sol") . So when i type in python3 deploy.py or python deploy.py it simply prints $compiled_sol

**Reply by oliveruf42** [reply]
Nevermind it works now! I think i had something mucked up with the deploy.py file so i went back and started from scratch with a fresh file. I setup a fresh virtual env and it finally worked! Thank you so much for helping :)

**Reply by cromewar** [reply]
I'm glad is solved now ;)

**Comment by caspixn** [comment]
Having a similar sort of problem. But I'm receiving....

Traceback (most recent call last):
  File "C:\Users\Will\Demos\web3_py_simple_storage\deploy.py", line 8, in <module>
    compiled_sol = compile_standard(
  File "C:\Users\Will\AppData\Local\Programs\Python\Python310\lib\site-packages\solcx\main.py", line 365, in compile_standard
    raise ContractsNotFound(
solcx.exceptions.ContractsNotFound: Input JSON does not contain any sources
> command: ``
> return code: `None`
> stdout:
None
> stderr:
None

**Reply by cromewar** [reply]
python 3.10 has problems with web3, use python 3.8 instead.

**Comment by mycrowprojects** [comment]
How the hell do you get off this list? (Email notifications for every comment made) omg.

I’ve unsubscribed 3? x’s

**Reply by cromewar** [reply]
Go to your github settings and remove the notifications from new content.

**Comment by mycrowprojects** [comment]
Right. I did that. But still woke up to 2 dozen emails. See attachment. Something is broken. 



Mike
::
MIKECHESSER-RØE
***@***.***
509-310-9445

“I never more than half succeed in expressing what I want to express. Actually not as much as that, but by no more than a tenth. That is still worth something. Often my writing is nothing but ‘stuttering.’” –Wittgenstein, Culture and Value, p. 18e

**Comment by maorazoulay** [comment]
Guys, only thing that worked for me was pretty simple: 

if you are on windows, just hold ctrl + shift + P and search for "python interpreter", then select the one that is recommended in the dropdown, that's it.

**Comment by kamal3951** [comment]
![Screenshot (1096)](https://user-images.githubusercontent.com/77496267/159451798-5070fd47-e264-4b38-b592-10e08ef3764f.png)
Hey, 3:44:00 I'm getting this error, It says solcx would not be resolves but I've tried installing the solcx by pip install py-solc-x

**Reply by cromewar** [reply]
@kamal3951 be sure you have selected the correct python interpreter on vscode.
